logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/featureConfig/SyncFeatureConfigsUseCase.kt

```diff
@@ -98,16 +99,14 @@ internal class SyncFeatureConfigsUseCaseImpl(
     }
 
     private fun handleFileSharingStatus(model: ConfigsStatusModel) {
-        if (kaliumConfigs.fileRestrictionEnabled) {
-            userConfigRepository.setFileSharingStatus(false, null)
-        } else {
-            val status: Boolean = model.status == Status.ENABLED
-            val isStatusChanged = when (isFileSharingEnabledUseCase().isFileSharingEnabled) {
-                null, status -> false
-                else -> true
-            }
-            userConfigRepository.setFileSharingStatus(status, isStatusChanged)
+        val status: Boolean = model.status == Status.ENABLED
```

Comment:
```suggestion
        val newStatus: Boolean = model.status == Status.ENABLED
```

So the logic can be better understood when reading:
`newStatus` is the thing we just received from the backend
`currentStatus` is the thing we have stored

Better to read instead of `state` vs `status`.